subprojects/logging/src/test/groovy/org/gradle/internal/logging/sink/OutputEventRendererTest.groovy

```diff
@@ -356,6 +363,44 @@ class OutputEventRendererTest extends OutputSpecification {
         outputs.stdOut.readLines() == ['info']
         outputs.stdErr == ''
     }
+
+    def "executor schedules at fixes rate"() {
+        given:
+        def executor = Mock(ScheduledExecutorService)
+        renderer.updateNowExecutor = executor
+
+        when:
+        renderer.addConsole(console, true, true, new UnixConsoleMetaData(true, true))
+
+        then:
+        1 * executor.scheduleAtFixedRate(_, _, _, TimeUnit.MILLISECONDS)
+    }
+
+    def "executor emits update now event when executing"() {
+        given:
+        def listener = new TestOutputEventListener()
+        renderer.addOutputEventListener(listener)
+        renderer.addConsole(console, true, true, new UnixConsoleMetaData(true, true))
+
+        when:
+        Thread.sleep(OutputEventRenderer.UPDATE_NOW_FLUSH_INITIAL_DELAY_MS + OutputEventRenderer.UPDATE_NOW_FLUSH_PERIOD_MS + 100)
```

Comment:
Look at how the tests were done in the other renderer, aka use catch the scheduled runnable and executed it when required as well as using `MockTimeProvider`.